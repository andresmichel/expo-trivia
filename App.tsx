import 'react-native-gesture-handler';
import React, { useState } from 'react';
import { StatusBar, AsyncStorage, YellowBox } from 'react-native';
import { Provider } from 'react-redux';
import { AppLoading } from 'expo';
import { Asset } from 'expo-asset';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { store } from './src/redux';
import { setColorScheme } from './src/redux/settings/actions';
import HomeScreen from './src/screens/home';
import QuizScreen from './src/screens/quiz';
import ResultsScreen from './src/screens/results';

YellowBox.ignoreWarnings(['Switch: `onTintColor`']);

const Stack = createStackNavigator();

const cacheAssetsAsync = () => {
    const images = [
        require('./assets/images/logo.png'),
    ];
    const cacheImages = images.map(image => {
        return Asset.fromModule(image).downloadAsync();
    });
    const checkStorageForSettings = async (key, callback) => {
        try {
            const value = await AsyncStorage.getItem(key);
            if (value !== null) store.dispatch(callback(value));
        } catch (error) {
            // 
        }
    }
    return Promise.all([
        ...cacheImages,
        checkStorageForSettings('colorScheme', value => setColorScheme(value)),
    ]);
}

function App() {
    const [isReady, setIsReady] = useState(process.env.NODE_ENV === 'test');
    if (!isReady) return (
        <AppLoading
            startAsync={async () => { cacheAssetsAsync() }}
            onFinish={() => setIsReady(true)}
            onError={console.warn}
        />
    );
    return (
        <Provider store={store}>
            <StatusBar hidden />
            <NavigationContainer>
                <Stack.Navigator headerMode={'none'} screenOptions={{ gestureEnabled: false }}>
                    <Stack.Screen name="Home" component={HomeScreen} />
                    <Stack.Screen name="Quiz" component={QuizScreen} />
                    <Stack.Screen name="Results" component={ResultsScreen} />
                </Stack.Navigator>
            </NavigationContainer>
        </Provider>
    );
}

export default App;