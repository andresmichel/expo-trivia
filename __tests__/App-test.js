import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App.tsx';

Date.now = jest.fn(() => 1584648513058);
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper')

describe('<App />', () => {
    it('renders correctly', () => {
        let root;
        renderer.act(() => {
            root = renderer.create(<App />);
        });
        expect(root.toJSON()).toMatchSnapshot();
    });
});