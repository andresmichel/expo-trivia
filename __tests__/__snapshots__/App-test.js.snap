// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "flex": 1,
                          },
                          Object {
                            "backgroundColor": "#F3F4FF",
                          },
                          Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "width": 240,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={false}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "transform": Array [
                                Object {
                                  "scale": AnimatedValue {
                                    "animatedStyles": Set {},
                                    "children": Array [],
                                    "done": false,
                                    "lastPosition": 0.8,
                                    "lastTime": undefined,
                                    "lastVelocity": undefined,
                                    "payload": undefined,
                                    "setValue": [Function],
                                    "startPosition": 0.8,
                                    "startTime": 1584648513058,
                                    "value": 1,
                                  },
                                },
                              ],
                            }
                          }
                        >
                          <Image
                            source={1}
                            style={
                              Object {
                                "height": 150,
                                "marginBottom": 20,
                                "width": 150,
                              }
                            }
                          />
                        </View>
                        <Text
                          inverted={false}
                          style={
                            Array [
                              Object {
                                "color": "#000",
                              },
                              Object {
                                "fontSize": 24,
                                "fontWeight": "bold",
                                "marginBottom": 40,
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          Welcome to the Trivia Challenge!
                        </Text>
                        <Text
                          inverted={false}
                          style={
                            Array [
                              Object {
                                "color": "#000",
                              },
                              Object {
                                "fontSize": 18,
                                "marginBottom": 20,
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          You will be presented with 10 True or False questions.
                        </Text>
                        <Text
                          inverted={false}
                          style={
                            Array [
                              Object {
                                "color": "#000",
                              },
                              Object {
                                "fontSize": 18,
                                "marginBottom": 20,
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          Can you score 100%?
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "position": "absolute",
                            "right": 20,
                            "top": 60,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          hitSlop={
                            Object {
                              "bottom": 5,
                              "left": 20,
                              "right": 20,
                              "top": 5,
                            }
                          }
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "borderRadius": 5,
                                "height": 50,
                                "justifyContent": "center",
                                "paddingHorizontal": 20,
                              },
                              false,
                              Object {
                                "backgroundColor": "#6D80FB",
                              },
                              Object {
                                "backgroundColor": "#F3F4FF",
                              },
                              false,
                              false,
                              Object {
                                "alignItems": "center",
                                "borderRadius": 20,
                                "height": 40,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "position": "absolute",
                              }
                            }
                          >
                            <Text />
                          </View>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "bottom": 60,
                            "maxWidth": 500,
                            "position": "absolute",
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          hitSlop={
                            Object {
                              "bottom": 5,
                              "left": 20,
                              "right": 20,
                              "top": 5,
                            }
                          }
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "borderRadius": 5,
                                "height": 50,
                                "justifyContent": "center",
                                "paddingHorizontal": 20,
                              },
                              false,
                              Object {
                                "backgroundColor": "#6D80FB",
                              },
                              false,
                              false,
                              false,
                              false,
                              Object {
                                "marginHorizontal": 20,
                              },
                            ]
                          }
                        >
                          <Text
                            inverted={true}
                            style={
                              Array [
                                Object {
                                  "color": "#FFF",
                                },
                                Object {
                                  "fontSize": 18,
                                  "fontWeight": "bold",
                                },
                              ]
                            }
                          >
                            Begin
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
